src/main/resources/codegen/java/ValueDataObject.ftl

```diff
@@ -34,4 +38,37 @@ public class ${dataValueObjectName} {
     return ${valueObjectName}.from(${valueObjectFields});
   }
 
-}
\ No newline at end of file
+  @Override
+  public int hashCode() {
+    return new HashCodeBuilder()
+              <#list memberNames as member>
+              .append(${member})
+              </#list>
+              .toHashCode();
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this){
+      return true;
+    }
+    if (!(obj instanceof ${valueObjectName})){
+      return false;
+    }
+    ${valueObjectName} another = (${valueObjectName}) obj;
+    return new EqualsBuilder()
```

Comment:
Seed with prime number `31`.

Reply:
For equals() or hashCode()? Why 31? HashCodeBuilder itself uses another prime number internally.

Reply:
@hurelhuyag https://stackoverflow.com/a/299748

Reply:
@VaughnVernon There is no difference between 37 and 31. Maybe 37 can be better than 31. But I changed it to 31.